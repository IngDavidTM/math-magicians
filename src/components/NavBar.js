import React from 'react';
import '../stylesheet/NavBar.css';
import { Link } from 'react-router-dom';

const NavBar = () => (
  <nav>
    <h1>
      Math Magicians
    </h1>
    <ul>
      <li><Link to="/">Home</Link></li>
      <li><Link to="/calculator">Calculator</Link></li>
      <li><Link to="/quote">Quote</Link></li>
    </ul>
  </nav>
);

export default NavBar;
