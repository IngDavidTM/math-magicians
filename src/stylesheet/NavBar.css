nav {
  display: flex;
  align-items: center;
  justify-content: space-between;
  background-color: #ffffffc5;
  backdrop-filter: blur(5px);
  box-shadow: 0 1px 7px #ccc;
}

nav h1 {
  font-size: 50px;
  padding: 20px;
  color: #6f9d50;
}

nav ul {
  display: flex;
  padding: 20px;
  color: #50959d;
}

nav li {
  list-style: none;
  padding: 15px;
}

nav li:nth-child(even) {
  border-left: 1px solid #ccc;
  border-right: 1px solid #ccc;
}
