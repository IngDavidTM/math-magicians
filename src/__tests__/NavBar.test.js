import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import NavBar from '../components/NavBar';

describe('test snapshop', () => {
  it('should render NavBar', () => {
    const renderNavBar = renderer
      .create(<BrowserRouter><NavBar /></BrowserRouter>)
      .toJSON();
    expect(renderNavBar).toMatchSnapshot();
  });
});
