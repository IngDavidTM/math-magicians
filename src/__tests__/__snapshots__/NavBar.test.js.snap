// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshop should render NavBar 1`] = `
<nav>
  <h1>
    Math Magicians
  </h1>
  <ul>
    <li>
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li>
      <a
        href="/calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
    </li>
    <li>
      <a
        href="/quote"
        onClick={[Function]}
      >
        Quote
      </a>
    </li>
  </ul>
</nav>
`;
